generators/pkg/writer/write.go

```diff
@@ -0,0 +1,41 @@
+// Package writer provides a method to write the generated task
+// to the yaml document.
+package writer
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"generators/pkg/generator"
+	"generators/pkg/parser"
+
+	"sigs.k8s.io/yaml"
+)
+
+// WriteToDisk writes the generated task
+// to the io.Writer
+func WriteToDisk(filename string, writer io.Writer) error {
+	file, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+
+	spec, err := parser.Parse(file)
+	if err != nil {
+		return err
+	}
+
+	task, err := generator.GenerateTask(spec)
+	if err != nil {
+		return err
+	}
+
+	data, err := yaml.Marshal(&task)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(writer, "%s", data)
```

Comment:
It's probably cleaner to use the writer interface instead. I don't know if there will be any weird edge cases in string conversion using `%s`.

e.g.
```go
_, err := writer.Write(data)
return err
```